core/src/test/java/com/novoda/merlin/service/MerlinServiceShould.java

```diff
@@ -112,6 +117,32 @@ public void callMerlinNetworkListenerOnDisconnectedWhenAConnectedConnectivityEve
         verify(rxCallbacksManager).onDisconnect();
     }
 
+    @Test
+    public void updateCurrentNetworkStatusRetrieverWhenSetHostNameIsCalled() {
+        String newTestHostName = "someNewTestHostName";
+        Context context = mock(Context.class);
+        ContextWrapper contextWrapper = mock(ContextWrapper.class);
+        ConnectivityManager connectivityManager = mock(ConnectivityManager.class);
+        Mockito.when(contextWrapper.getApplicationContext()).thenReturn(context);
+        Mockito.when(context.getSystemService(Context.CONNECTIVITY_SERVICE)).thenReturn(connectivityManager);
```

Comment:
It'd be nice if this could be a static import 😉  and same for the assertion https://github.com/novoda/merlin/pull/84/files#diff-041de7855cd1408bc7e88f09d0371af3R143